skademlia/table.go

```diff
@@ -110,11 +110,19 @@ func (t *Table) Update(target *ID) error {
 	if found := t.Find(b, target); found != nil {
 		b.Lock()
 
-		// address might differ for same public key (checksum
 		id := found.Value.(*ID)
-		id.address = target.address
+		// address might differ for same public key (checksum)
+		if id.address != target.address {
```

Comment:
I think we can just replace the whole code with:

```
b.Remove(found)
b.PushFront(target)
```

The if statement in this case is redundant.

Reply:
No problem then, let's just leave it as it is.